docs/src/userguide/language_basics.rst

```diff
@@ -538,8 +538,71 @@ Like other Python looping statements, break and continue may be used in the
 body, and the loop may have an else clause.
 
 
-The include statement
-=====================
+Cython file types
+=================
+
+There are three file types in Cython:
+
+* The implementation files, carrying a ``.pyx`` suffix.
+* The definition files, carrying a ``.pxd`` suffix.
+* The include files, carrying a ``.pxi`` suffix.
+
+The implementation file
+-----------------------
+
+The implementation file, as the name suggest, contains the implementation
+of your functions, classes, extension types, etc. Nearly all the
+python syntax is supported in this file. Most of the time, a ``.py``
+file can be converted into a ``.pyx`` file without changing
+any code, and Cython will retain the python behavior, with a
+slight speed boost.
```

Comment:
- "renamed" is better than "converted" here
- there should be no performance difference between compiled .py files and .pyx files
- I think we should not encourage users to rename their files without reason and instead just make it clear that .pyx files allow (not enforce!) extended Cython syntax whereas .py files do not and strictly require Python syntax. That is the essential difference.